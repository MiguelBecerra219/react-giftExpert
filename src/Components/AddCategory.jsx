import { useState } from 'react'
import PropTypes from 'prop-types'

export const AddCategory = ({ onNewCategory }) => {
  const [inputValue, setinputValue] = useState('')
  const onInputChange = (event) => {
    setinputValue(event.target.value)
  }

  const onSubmit = (event) => {
    event.preventDefault()
    if (inputValue.trim().length <= 1) return
    onNewCategory(inputValue.trim())
    setinputValue('')
  }

  return (
    <form onSubmit={onSubmit /* llamar la funcion de esta manera automaticamente pasa el evento como prop */} aria-label='form' >
      <input
        type="text"
        placeholder="Buscar gift"
        value={inputValue}
        onChange={onInputChange /* llamar la funcion de esta manera automaticamente pasa el evento como prop */}
      />
    </form>
  )
}

AddCategory.proptype = {
  onNewCategory: PropTypes.func.isRequired
}
