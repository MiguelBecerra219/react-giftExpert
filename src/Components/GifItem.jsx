import PropTypes from 'prop-types'

export const GifItem = ({ title, url }) => {
  return (
  <article className="card">
    <h2>{title}</h2>
    <img src={url} alt={title}></img>
  </article>
  )
}

GifItem.protoTypes = {
  title: PropTypes.string.isRequireds,
  url: PropTypes.string.isRequireds
}
