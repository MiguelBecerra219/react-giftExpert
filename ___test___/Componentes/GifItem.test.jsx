import { render, screen } from '@testing-library/react'
import { GifItem } from '../../src/Components/GifItem'

/* eslint-disable */
describe('Pruebas al componente GifItem', () => { 
    const title = 'Titulo Gif'
    const url = 'https://media2.giphy.com/media/IvOFcGZeDA76P6XryO/giphy.gif?cid=377eb6f21m3n5qkuxxtw5nu6pq39v7g1s3m9imdaayx8j3yb&ep=v1_gifs_search&rid=giphy.gif&ct=g'

    test('La renderizacion debe coincidir con el snapShop', () => { 
        const { container } = render(<GifItem title={ title } url={ url }/>)
        expect(container).toMatchSnapshot()
        
     })

     test('debe de mostrar la imagen con el url y el alt indicado', () => { 
        render(<GifItem title={ title } url={ url }/>)
        screen.debug()

        const { src, alt } = screen.getByRole('img')

        expect(src).toBe(url)
        expect(alt).toBe(title)
      })

      test('debe mostrar el titulo en el componente', () => { 
        render(<GifItem title={ title } url={ url }/>)
        expect(screen.getByText(title)).toBeTruthy()
       })
 })