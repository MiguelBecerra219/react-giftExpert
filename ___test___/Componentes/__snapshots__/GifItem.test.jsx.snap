// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas al componente GifItem La renderizacion debe coincidir con el snapShop 1`] = `
<div>
  <article
    class="card"
  >
    <h2>
      Titulo Gif
    </h2>
    <img
      alt="Titulo Gif"
      src="https://media2.giphy.com/media/IvOFcGZeDA76P6XryO/giphy.gif?cid=377eb6f21m3n5qkuxxtw5nu6pq39v7g1s3m9imdaayx8j3yb&ep=v1_gifs_search&rid=giphy.gif&ct=g"
    />
  </article>
</div>
`;
